src/puppetlabs/pcp/client.clj

```diff
@@ -285,11 +286,15 @@
   (let [{:keys [should-stop websocket-client websocket-connection]} client]
     ;; NOTE:  This true value is also the sentinel for make-connection
     (deliver should-stop true)
-    (if (-connected? client)
-      (try
-        (ws/close @@websocket-connection)
-        (catch java.util.concurrent.ExecutionException exception
-          (log/debug exception (i18n/trs "exception while closing the connection; connection already closed")))))
+    ;; Make access to websocket-connection atomic, so we don't cause an exception when
+    ;; racing during shutdown. It's possible for the connection to be closed between
```

Comment:
Last sentence might be slightly confusing. It's that "the websocket-connection atom could be reset to true".